powerline/bindings/lemonbar/powerline-lemonbar-multimonitor.py

```diff
@@ -0,0 +1,94 @@
+#!/usr/bin/env python
+# vim:fileencoding=utf-8:noet
+from __future__ import (unicode_literals, division, absolute_import, print_function)
+import sys
+import time
+import re
+import subprocess
+
+from threading import Lock, Timer
+from argparse import ArgumentParser, REMAINDER
+
+from powerline.lemonbar import LemonbarPowerline
+from powerline.lib.monotonic import monotonic
+from powerline.lib.dict import mergedicts
+
+
+if __name__ == '__main__':
+	parser = ArgumentParser(
+		description='Powerline BAR bindings.',
+		usage='%(prog)s [-h] [--i3] [--height HEIGHT] [--theme THEME [THEME ...]] -- args')
+	parser.add_argument(
+		'--i3', action='store_true',
+		help='Subscribe for i3 events.'
+	)
+	parser.add_argument(
+		'--height', default='',
+		help='Bar height'
+	)
+	parser.add_argument(
+		'--theme', '-t',
+		default=[], nargs='+',
+		help='makes all outputs that containing a string use the specified theme. Format like "out=theme"'
+	)
+	parser.add_argument(
+		'args', nargs=REMAINDER,
+		help='Extra arguments for lemonbar'
+	)
+	args = parser.parse_args()
+	theme_rules = {}
+	for theme in args.theme:
+		out, theme = theme.split('=', 1)
+		theme_rules[out] = theme
+
+	active_screens = [match.groupdict() for match in re.finditer(
+		'^(?P<name>\w+) connected (?P<width>\d+)x(?P<height>\d+)\+(?P<x>\d+)\+(?P<y>\d+)',
+		subprocess.check_output(['xrandr', '-q']).decode(),
+		re.MULTILINE
+	)]
+
+	powerlines = []
+
+	for screen in active_screens:
+		theme = None
+		for out in theme_rules:
+			if out in screen['name']:
+				theme = theme_rules[out]
+
+		line = LemonbarPowerline(theme=theme, output=screen['name'])
+		command = ['bar-aint-recursive', '-g', '{}x{}+{}'.format(screen['width'], args.height, screen['x'])] + args.args[1:]
+		process = subprocess.Popen(command, stdin=subprocess.PIPE)
+		powerlines.append((line, process, int(screen['width']) / 5))
+
+	lock = Lock()
+	modes = ['default']
+
+	def render(reschedule=False):
+		if reschedule:
+			Timer(0.5, render, kwargs={'reschedule': True}).start()
```

Comment:
@S0lll0s For per-monitor theme configuration there *is* facility: it is local themes. In case somebody needs overriding its configuration the whole thing needs to be written to accept overrides from environment variables because this is *consistent* with other bindings which do not have something more usable (namely shell and tmux, vim and ipython are configured using existing vim and ipython own configuration facilities) and also more flexible. Though you will have to inherit from `ShellPowerline` then and use `powerline.commands.main.finish_args`. It makes sense to move https://github.com/ZyX-I/powerline/blob/4382993d43057ed46bc5d6cc1f18affb5a0e6d39/powerline/commands/main.py#L116-L134 and https://github.com/ZyX-I/powerline/blob/4382993d43057ed46bc5d6cc1f18affb5a0e6d39/powerline/commands/main.py#L144-L149 to a separate function for this purpose, because shell also allows overriding from command-line (though this was officially deprecated, I am not going to remove it ever).